doc/source/tutorials/basictutorial.rst

```diff
@@ -0,0 +1,656 @@
+
+=====================
+``if`` vs ``else if``
+=====================
+
+Hopefully you now know the basics of how ``if`` works. You might be wondering why use ``else if`` if it's the same as ``if``.
+
+Using ``else if``
+_________________
+::
+
+	if distance <= 1 {
+  	  print "Distance is within a meter.".
+	} else if distance <= 100 {
+  	  print "Distance is within 100 meters.".
+	} else {
+	  print "Distance is farther than 100 m.".
+	}
+
+Only using ``if``
+__________________
+::
+
+	if distance <= 1 {
+	  print "Distance is within a meter.".
+	}
+	if distance <= 100 {
+  	  print "Distance is within 100 meters.".
+	}
+	if distance > 1000 {
+  	  print "Distance is farther than 1 kilometer.".
+	}
+
+We now run into some problems: if your distance is less than a meter you'll get the following messages: ::
+
+	Distance is within a meter.
+	Distance is within 100 meters.
+	Distance is farther than 1 kilometer.
+
+As you can imagine this isn't good. If we're at less than a meter away from something and the commands for if we would be farther
+than 1 kilometer show up we have a big problem. This could be fixed by doing the following, but **THIS IS UNNECESSARILY** complex: ::
+
+ set Done to false.
+
+ if Done = false {
+   if distance <= 1 {
+     print "Distance is within a meter.".
+     set NotDone to false.
+   }
+ }
+
+ if Done = false {
+   if distance <= 100 {
+     print "Distance is within 100 meters.".
+     set NotDone to false.
+   }
+ }
+
+ if Done = false {
+   if distance > 1000 {
+     print "Distance is farther than 1 kilometer.".
+     set NotDone to false.
+   }
+ }
+
+Now this essentially does the same as the ``else if`` script but it's way more confusing and complicating.
+
+====================
+Until, lock and wait
+====================
+
+The ``wait`` command pretty straight forward: ::
+
+  wait 10.
+ print "done waiting".
+
+
+It will take 10 seconds before ``done waiting`` shows up.
+Using ``wait 0`` will let the script wait for one physics tick, this can be handy for
+when you're doing stuff with maneuvers. Maneuvers don't show up instantly but show up
+after one physics tick. More about maneuvers in part ???.
+
+The ``until`` command will keep looping a piece of code until the given value has been met.
+Before we can talk about until loops let's first talk about ``time:seconds`` and the ``lock`` command. ::
+
+  print time:seconds.
+
+
+will print the current time in seconds. Let's say the time is 1 minute.
+It would print ``60``. You can also set the current time as a variable: ::
+
+  set CurrentTime to time:seconds.
+
+The variable ``CurrentTime`` will stay 60 seconds. Using the ``set`` command will look at
+a value and pick that value to stay the  same, even if the value it was set to changes.
+For instance, printing ``CurrentTime`` would give 60. Not only at ``time:seconds`` = 60,
+also at any other time like ``time:seconds`` = 4000. Eventhough ``time:seconds`` is 4000,
+``CurrentTime`` is still 60.
```

Comment:
This is really really unclear.  I *know* the language and had to look at this several times to figure out what you were trying to say.  You are contrasting ``set`` to ``lock``, taking time to emphasize the point that ``set`` only assigns the value once and then it stays the same after that, but you haven't introduced ``lock`` yet, so you are drawing a comparison that doesn't mean anything *yet* to the reader
A better example would be something a bit more realistic - like this:

"Consider what would happen if you ran this (type it in and try it):"
```
print time:seconds.
wait 1.
print time:seconds.
wait 1.
print time:seconds.
```
"Notice it prints 3 different numbers, because you waited a second between each print statement, time:seconds was different each time."

"Now compare that with what happens if you set it into a variable first, then print that variable:"
```
set CurrentTime to time:seconds.
print CurrentTime.
wait 1.
print CurrentTime.
wait 1.
print CurrentTime.
```
"Notice that CurrentTime prints the same number each time, even though several seconds have passed.  This is because the ``set`` command stores the result *at the time it was run* into the variable, and the value stays the same until you set it again with another ``set`` command."